public/main.ts

```diff
@@ -132,27 +132,196 @@ const kingdoms = [
 const player = kingdoms[0];
 player.farms.set(Crop.Rye, new Farm(getRandInt(minStartingRye, maxStartingRye)));
 
+// indicies for select fields
+let selectedFarm = "Rye";
+// let selectedCrop = "Rye";
+// let sellAmount = "";
+
 function promptPlayer(message: string): string {
     let result = null;
     while(result == null)
         result = window.prompt(message, "");
     return result;
 }
 
-function updateText() {
-
+interface Printable { toString(): string }
+type tag = HTMLElement | HTMLInputElement;
+function setElementInnerHTML(el: tag, content: Printable) {
+    el.innerHTML = content.toString();
 }
+function setIdInnerHTML(id: string, content: Printable) {
+    document.getElementById(id)!.innerHTML = content.toString();
+}
+
+
+function handleBuy(ev: MouseEvent) {
+    const name = (ev.target as HTMLElement).id;
+    const key = name.substring(0, name.indexOf("-"));
+    const crop = Crop[key as keyof typeof Crop];
+
+    if(player.gold >= cropMarket.get(crop)!) {
+
+        if(!player.farms.has(crop)) {
+            player.farms.set(crop, new Farm(1));
+            const stats = document.getElementById("gold-grain-stats");
+            const grainStat = document.createElement("p");
+            grainStat.id = Crop[crop] + "-stats";
+            setElementInnerHTML(grainStat, player.farms.get(crop)!.stockpile.toString() + " " + Crop[crop]);
+
+            const farms = document.getElementById("farms");
+            const farmOption = document.createElement("option");
+            farmOption.id = Crop[crop] + "-farm-option";
+            setElementInnerHTML(farmOption, Crop[crop]);
+            farms!.appendChild(farmOption);
 
+            const sellCrops = document.getElementById("crops-sale");
+            const cropOption = document.createElement("option");
+            cropOption.id = Crop[crop] + "-crop-option";
+            setElementInnerHTML(cropOption, Crop[crop]);
+            sellCrops!.appendChild(cropOption);
 
-window.setInterval(() => {
-    timer += tickRate;
+            stats?.appendChild(grainStat);
+        }
+        else player.farms.get(crop)!.stockpile += 1;
 
+        player.gold-= cropMarket.get(crop)!;
+    }
+
+
+}
+
+// load gui
+function loadGUI() {
+    cropMarket.forEach((value, key, map) => {
```

Comment:
The map parameter is unused. Please remove
```suggestion
    cropMarket.forEach((value, key) => {
```

Reply:
This occurs in a few other places.